service/history/workflow/mutable_state_impl.go

```diff
@@ -1347,7 +1349,9 @@ func (ms *MutableStateImpl) GetWorkflowCloseTime(ctx context.Context) (time.Time
 		}
 		return completionEvent.GetEventTime().AsTime(), nil
 	}
-
+	if ms.executionInfo.CloseTime == nil {
+		return time.Time{}, ErrMissingCloseTimeInfo
+	}
```

Comment:
This is a breaking change. Let's not do this.

Reply:
ah that is right. It will return error if I have this code block. I removed them